nano/node/websocket.hpp

```diff
@@ -63,6 +64,28 @@ namespace websocket
 		message block_confirmed (std::shared_ptr<nano::block> block_a, nano::account const & account_a, nano::amount const & amount_a, std::string subtype);
 	};
 
+	/** Filtering options for subscriptions */
+	class options
+	{
+	public:
+		virtual bool filter (boost::property_tree::ptree const & message_a, nano::node & node_a) const
+		{
+			return false;
+		}
+	};
+
+	class confirmation_options final : public options
+	{
+	public:
+		confirmation_options ();
+		confirmation_options (boost::property_tree::ptree const & options_a);
+		virtual bool filter (boost::property_tree::ptree const & message_a, nano::node & node_a) const;
+
+	private:
+		bool all_local_accounts{ false };
+		std::unordered_set<nano::account> accounts;
```

Comment:
Would it be better to have these as `std::strings` and not do the account decoding during `filter`?

Reply:
Yeah that makes sense.